lib/mongo/address.rb

```diff
@@ -146,7 +152,7 @@ def inspect
     #
     # @since 2.0.0
     def socket(socket_timeout, ssl_options = {})
-      @resolver ||= initialize_resolver!(ssl_options)
+      @resolver = initialize_resolver!(ssl_options)
       @resolver.socket(socket_timeout, ssl_options)
```

Comment:
Since the resolver is now local, it should no longer be an instance variable (remove `@`).

Reply:
Hmm, looks like some tests fail when I do that. I'll look into what it takes to fix them.